source/NuLink.Cli/INuLinkCommand.cs

```
@@ -2,6 +2,6 @@ namespace NuLink.Cli
 {
     public interface INuLinkCommand
     {
-        int Execute(NuLinkCommandOptions options);
+        void Execute(NuLinkCommandOptions options);
```

The idea behind the `int` return value is to return an exit code, which can be useful for scripting. I guess `int` becomes tricky because of a multi-package operation, which can result in a mix of successes and failures. In such a case, I'd prefer to define the overall success and failure.

This is what I suggest:
- For `link` command, the "package already linked" situation, when the link target is the desired location, should be a success rather than an error (print a success message, do nothing).
- For `unlink` command, if the package is not linked, it's a success and not an error (print a success message, do nothing).
- Continue to handle all other errors
- Let the user choose one of two modes:
  - Fail fast (default): stop on the first error and exit with an error code
  - Ignore errors: if an `--ignore-errors` flag is specified, we run all packages, report all errors as warnings, and always exit with code 0.

(as for implementation, see comment below about exceptions)